elastic/elastic-agent
# Add logging to EventLog for error, install, uninstall

## What does this PR do?

On Windows, if the `install`, `uninstall`, or `enroll` commands fail, they will now attempt to add the error to the Windows Application Event Log.  This is the same behavior that the `run` command has.

## Why is it important?

If any of these commands are done with automation it can be difficult to capture the errors that appear on standard error.  This gives us a chance of capturing the error and reviewing later.

## Checklist




## Disruptive User Impact



## How to test this PR locally

1. Build with this PR
2. Run Install
3. Try to uninstall with bad arguments or a file open in the install directory
4. Error should be visible in the Event Viewer

## Related issues


- Closes #6338

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
